File: leetcode/242. Valid Anagram.py
class Solution(object):
	def isAnagram(self, s, t):
		if len(s) == len(t):
			return self.statistic(s) == self.statistic(t)
		else:
			return False
	
	def statistic(self, s):
		ret = {}
		for c in s:
			if ret.get(c) != None:
				ret[c] += 1
			else:
				ret[c] = 1
		return ret
		
Solution().isAnagram("hi", "ih");

File: leetcode/242. Valid Anagram.cc
// Copyright 2017 Qi Wang
// Date: 2017-07-24
class Solution {
 public:
  bool isAnagram(string s, string t) {
    if (s.size() != t.size()) return false;
    int cnt[26];
    memset(cnt, 0, sizeof cnt);
    for (char c : s) {
      ++cnt[c - 'a'];
    }
    for (char c : t) {
      --cnt[c - 'a'];
    }
    return count(cnt, cnt + 26, 0) == 26;
  }
};
